internal/configs/configurator.go

```diff
@@ -454,6 +468,86 @@ func (cnf *Configurator) AddOrUpdateVirtualServers(virtualServerExes []*VirtualS
+		newUpstreams[u.Name] = true
+		newUpstreamsNames = append(newUpstreamsNames, u.Name)
+
+		for _, server := range u.Servers {
+			podInfo := transportServerEx.PodsByIP[server.Address]
+			labelKey := fmt.Sprintf("%v/%v", u.Name, server.Address)
+			upstreamServerPeerLabels[labelKey] = []string{podInfo.Name}
+
+			if cnf.staticCfgParams.NginxServiceMesh {
+				ownerLabelVal := fmt.Sprintf("%s/%s", podInfo.OwnerType, podInfo.OwnerName)
+				upstreamServerPeerLabels[labelKey] = append(upstreamServerPeerLabels[labelKey], ownerLabelVal)
+			}
+
+			newPeers[labelKey] = true
+			newPeersIPs = append(newPeersIPs, labelKey)
+		}
+	}
+
+	key := fmt.Sprintf("%v/%v", transportServerEx.TransportServer.Namespace, transportServerEx.TransportServer.Name)
+
+	removedPeers := findRemovedKeys(cnf.metricLabelsIndex.transportServerUpstreamPeers[key], newPeers)
+	cnf.metricLabelsIndex.transportServerUpstreamPeers[key] = newPeersIPs
+	cnf.latencyCollector.UpdateUpstreamServerPeerLabels(upstreamServerPeerLabels)
+	cnf.latencyCollector.DeleteUpstreamServerPeerLabels(removedPeers)
+
+	removedUpstreams := findRemovedKeys(cnf.metricLabelsIndex.transportServerUpstreams[key], newUpstreams)
+	cnf.latencyCollector.UpdateUpstreamServerLabels(labels)
+	cnf.metricLabelsIndex.transportServerUpstreams[key] = newUpstreamsNames
+
+	cnf.latencyCollector.DeleteUpstreamServerLabels(removedUpstreams)
+	cnf.latencyCollector.DeleteMetrics(removedPeers)
+
+	if cnf.isPlus {
+		cnf.labelUpdater.UpdateStreamUpstreamServerPeerLabels(upstreamServerPeerLabels)
+		cnf.labelUpdater.DeleteStreamUpstreamServerPeerLabels(removedPeers)
+		cnf.labelUpdater.UpdateStreamUpstreamServerLabels(labels)
+		cnf.labelUpdater.DeleteStreamUpstreamServerLabels(removedUpstreams)
+
+		streamServerZoneLabels := make(map[string][]string)
+		newZones := make(map[string]bool)
+		newZonesNames := []string{transportServerEx.TransportServer.Name}
+
+		streamServerZoneLabels[transportServerEx.TransportServer.Name] = createStreamServerZoneLabels(
+			"transportserver", transportServerEx.TransportServer.Name, transportServerEx.TransportServer.Namespace)
+
+		newZones[transportServerEx.TransportServer.Name] = true
+		removedZones := findRemovedKeys(cnf.metricLabelsIndex.transportServerServerZones[key], newZones)
+		cnf.metricLabelsIndex.transportServerServerZones[key] = newZonesNames
+		cnf.labelUpdater.UpdateStreamServerZoneLabels(streamServerZoneLabels)
+		cnf.labelUpdater.DeleteStreamServerZoneLabels(removedZones)
+	}
+}
+
+func (cnf *Configurator) deleteTransportServerMetricsLabels(key string) {
+	cnf.latencyCollector.DeleteUpstreamServerLabels(cnf.metricLabelsIndex.transportServerUpstreams[key])
+	cnf.latencyCollector.DeleteUpstreamServerPeerLabels(cnf.metricLabelsIndex.transportServerUpstreamPeers[key])
+	cnf.latencyCollector.DeleteMetrics(cnf.metricLabelsIndex.transportServerUpstreamPeers[key])
+
+	if cnf.isPlus {
```

Comment:
this will always be true, because we only need to update labels for NGINX Plus.  could you remove this `if`?